### PR TITLE
Development branch for version 3.0.0

### LINKED ISSUE #121: Add Oculus Quest 2 support

### PR DESCRIPTION
**Pull request by m4gr3d**
### Features:

- Update to version **23** of the Oculus Mobile SDK.
- Add support for the Oculus Quest 2
- Add `OvrSystem` API:
  - `OvrSystem::is_oculus_quest_1_device()`: Use to check for Oculus Quest 1 devices
  - `OvrSystem::is_oculus_quest_2_device()`: Use to check for Oculus Quest 2 devices
  - `OvrSystem::get_device_type()`: Return an `int` describing the device type. The returned value can be checked against `OvrVrApiTypes.OvrDeviceType` enum values.
- Update the demo project with support for the Oculus Quest 2
- Bug fixes and documentation improvement

### Breaking Changes:

- Deprecate support for the Oculus Go and clean up deprecated APIs

Fix #121

### CODE REVIEW

**Comment by NeoSpark314** [comment]
Code changes look good; nice work!
Build workd fine but when I try to deploy the demo application I get the following error:
```
Exception occurred while dumping:
java.lang.IllegalArgumentException: Unknown package: org.godotengine.demo.vr.oculus.mobile
        at com.android.server.pm.Settings.isOrphaned(Settings.java:4135)
        at com.android.server.pm.PackageManagerService.isOrphaned(PackageManagerService.java:18500)
        at com.android.server.pm.PackageManagerService.deletePackage(PackageManagerService.java:15895)
        at com.android.server.pm.PackageInstallerService.uninstall(PackageInstallerService.java:889)
        at com.android.server.pm.PackageManagerShellCommand.runUninstall(PackageManagerShellCommand.java:792)
        at com.android.server.pm.PackageManagerShellCommand.onCommand(PackageManagerShellCommand.java:118)
        at android.os.ShellCommand.exec(ShellCommand.java:98)
        at com.android.server.pm.PackageManagerService.onShellCommand(PackageManagerService.java:18760)
        at android.os.Binder.shellCommand(Binder.java:509)
        at android.os.Binder.onTransact(Binder.java:408)
        at android.content.pm.IPackageManager$Stub.onTransact(IPackageManager.java:2399)
        at com.android.server.pm.PackageManagerService.onTransact(PackageManagerService.java:3153)
        at android.os.Binder.execTransact(Binder.java:606)
Installing to device (please wait...): Oculus Quest
Performing Streamed Install
adb: failed to install C:\Users\neosp\AppData\Local\Temp/Godot/tmpexport.apk: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Package /data/app/vmdl741549785.tmp/base.apk has no certificates at entry AndroidManifest.xml]
```
Maybe it is this a setup error on my side?

File: plugin/src/main/assets/addons/godot_ovrmobile/OvrVrApiTypes.gd [hunk 1]
```diff
@@ -1,36 +1,13 @@
...
-	VRAPI_DEVICE_TYPE_OCULUSGO_START		= 64,
-	VRAPI_DEVICE_TYPE_OCULUSGO				= 64, #VRAPI_DEVICE_TYPE_OCULUSGO_START,
-	VRAPI_DEVICE_TYPE_MIVR_STANDALONE		= 64 + 1, #VRAPI_DEVICE_TYPE_OCULUSGO_START + 1,	#< China-only SKU
-	VRAPI_DEVICE_TYPE_OCULUSGO_END			= 127,
-
 	VRAPI_DEVICE_TYPE_OCULUSQUEST_START		= 256,
 	VRAPI_DEVICE_TYPE_OCULUSQUEST			= 256 + 3, #VRAPI_DEVICE_TYPE_OCULUSQUEST_START + 3,
 	VRAPI_DEVICE_TYPE_OCULUSQUEST_END		= 319,
+	VRAPI_DEVICE_TYPE_OCULUSQUEST2_START = 320,
+	VRAPI_DEVICE_TYPE_OCULUSQUEST2 = VRAPI_DEVICE_TYPE_OCULUSQUEST2_START,
```
Comment (NeoSpark314) [inline plugin/src/main/assets/addons/godot_ovrmobile/OvrVrApiTypes.gd:9]: Is VRAPI_DEVICE_TYPE_OCULUSQUEST2_START defined somewhere?

Comment (NeoSpark314) [reply plugin/src/main/assets/addons/godot_ovrmobile/OvrVrApiTypes.gd:9]: Ah; it is defined above; but for me GDScript did not resolve this...

Comment (NeoSpark314) [reply plugin/src/main/assets/addons/godot_ovrmobile/OvrVrApiTypes.gd:9]: I get an error `Expected a constant expression` for this line

Comment (m4gr3d) [reply plugin/src/main/assets/addons/godot_ovrmobile/OvrVrApiTypes.gd]: @NeoSpark314 I fixed the issue!

**Review by NeoSpark314 — APPROVED** [review]

**Pull request merged**